[MB-7020][MB-7021] Fix address fieldset component

suzubara opened the pull request:
## Description

This PR started as just a story to create a reusable component for the Service Member mailing address fields, but as I started creating an example of that in Storybook I realized there were issues with how the `AddressFields` were using both Formik's `Field` component _and_ the MilMove `TextInput` (which uses Formik's `useField` hook). This was resulting in some very confusing and broken code, so I used this branch to clean up & add clarification to the various ways to use `TextInput` and some related components.

To complete the original story AC, I added examples of how to use the existing `AddressFields` component to render the service member address fields (see **Components / Fieldsets / AddressFields / Current Residential Address, With Additional Text** in Storybook). Since this requires minimal code on top of AddressFields, I don't think there's any reason to create a new component for it.

As of this PR, if you want to add a text input to a Formik form you can use the `TextField` component, which is nearly identical to the formerly named `TextInput` component. Since this component uses `useField`, you should _not_ pass a `value` prop to it - this will conflict with the `value` prop that Formik will provide.

- I renamed `TextInput` to `TextField` and simplified the component code. It renders ReactUSWDS's TextInput inside a FormGroup with a label, hint, error, warning as it did before. I also removed the separate `TextInputMinimal` component because it was never actually used outside of `TextField`, and was also duplicating `useField`.
- I moved `TextInputMasked` to its own file and renamed to `MaskedTextField`. I made no code changes.
- I applied changes to both `AddressFields` and `ContactInfoFields`, and the only place in the application where these are used is the `MtoShipmentForm` so I made some required changes there as well. Changes were removing the `values` prop (since this will be passed in via `useField`) and removing the use of Formik's `<Field />` component.
- I added some tests and also converted some existing tests to use RTL instead of Enzyme

## Reviewer Notes

For the story AC, verify the Current Residential Address Storybook example matches the "Current mailing address" block, and the With Additional Text example matches the "Backup mailing address" block in the Zeplin design ([desktop](https://zpl.io/29X40Pd)) ([mobile](https://zpl.io/b6XZ7mg))

There are no expected user-facing changes, but it's worth spot checking forms that use the changed components:
- Customer / MTO Shipment Form (add/edit shipments)
- Office / Allowances Detail Form
- Office / Orders Detail Form
- Office / Reject Service Item modal

## Setup

To run the app locally:
```sh
make server_run
make client_run
```

You can also view the AddressFields component in Storybook:
```
yarn storybook
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [MB-7020](https://dp3.atlassian.net/browse/MB-7020) & [MB-7021](https://dp3.atlassian.net/browse/MB-7021)

## Screenshots

![image](https://user-images.githubusercontent.com/2723066/111370500-51acd700-8666-11eb-8c66-6a302dca04b4.png)

![image](https://user-images.githubusercontent.com/2723066/111370532-5b363f00-8666-11eb-84eb-62cf862b15f3.png)

![image](https://user-images.githubusercontent.com/2723066/111377103-48276d00-866e-11eb-9cdf-5486db7e0763.png)

![image](https://user-images.githubusercontent.com/2723066/111377178-583f4c80-866e-11eb-8208-b1c2288aa453.png)

robot-mymove commented:
<!--
  0 failure: 
  1 warning:  It looks like you...
  1 messages
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  It looks like you are attempting to bypass a linter rule, which is not within
      security compliance rules.
** Contains a rule that is not in the permitted eslint list. You are free to disable only: (
no-underscore-dangle
,prefer-object-spread
,object-shorthand
,camelcase
,react/jsx-props-no-spreading
,react/destructuring-assignment
,react/forbid-prop-types
,react/prefer-stateless-function
,react/sort-comp
,import/no-extraneous-dependencies
,import/order
,import/prefer-default-export
,import/no-named-as-default
) **
 Please remove the bypass code and address the underlying issue. cc: @transcom/Truss-Pamplemoose
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:link: <a href="https://dp3.atlassian.net/browse/MB-7020">MB-7020</a></td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0414a0c5837dfb9742c25108b3ff2f3ad1c155ef
</p>

lynzt reviewed (approved):
LGTM 🚢

suzubara commented on src/components/form/fields/MaskedTextField.test.jsx:
```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
```
I was just renaming the file without making any code changes

amanda5iu reviewed (commented):
This looks great Suz! Can we try your suggestion to add padding between the fields so that they aren't squished in the error state? I think it's ok if adjacent fields are misaligned for a little bit.

Shkeating commented:
from a design standpoint, only thing I see off is the space above the signature on the `Now for the official part`field, which should be 24px as per [the designs on Zeplin](https://zpl.io/VDeo0xl): 
<img width="619" alt="image" src="https://user-images.githubusercontent.com/59394696/111546026-6617cf00-874d-11eb-9058-95a2fde9e168.png">

YanZ777 reviewed (approved):
From a code standpoint, the changes look good to me!

suzubara commented:
@amanda5iu @Shkeating Ok, I think both issues have been resolved! The margin in the Agreement form should be fixed: 
![image](https://user-images.githubusercontent.com/2723066/111656512-9789ac80-87d8-11eb-8ea7-b319b16a1f65.png)

and I updated the error styles so they're in line with the USWDS defaults (we were overwriting a margin that I don't think we needed to). I also added a Storybook example called "Inside Section Wrapper" to show how the form would look when on the page: 
![image](https://user-images.githubusercontent.com/2723066/111656807-dd467500-87d8-11eb-8774-28fa40517b4b.png)

Shkeating reviewed (approved):
changes look good!! 😎✅

Shkeating commented:
also, hooray to all the pages getting tracked in storybook!! 🥳

I did notice on your Kitchen sink form page, that this checkbox could use some space above it.

<img width="438" alt="image" src="https://user-images.githubusercontent.com/59394696/111687116-4beaf900-8800-11eb-9399-b9e88e1c28dc.png">

amanda5iu reviewed (approved):
Looks so good! 🎉
